src/garage/replay_buffer/replay_buffer.py

```diff
@@ -12,173 +12,75 @@
 import abc
 from abc import abstractmethod
 
-import numpy as np
-
 
 class ReplayBuffer(metaclass=abc.ABCMeta):
```

Comment:
Immediately I see an issue here where there is 2 interfaces: a `TimeStep`-oriented one and `Trajectory`-oriented one.

Do we want to:
* Force all replay buffers to implement both semantics?
* Standardize on adding/sampling time steps?
* Standardize on adding/sampling trajectories? Note this would make single-stepping sampling a pain.
* Choose one interface as the universal case, define a more-specialized interface for the less-universal case? In this case, I think `TimeStep`-oriented is the universal case and `Trajectory`-oriented is the specific case.
* Define two different interfaces, entirely.